### PR TITLE
Creates a node compatibility mode

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
By default this will install node v20 LTS, but if a user is detected to be using win 7 it's node v14

This lets us run higher node versions (with presumably more stable and performant content) while allowing win 7 users to play

I should note that this is making clean tgui builds run at ~6.7sec which is about a 6.9% speed increase (nice) from the previous #80310
## Why It's Good For The Game
Better tools
## Changelog
N/A nothing player facing

### CODE REVIEW

File: tgui/.prettierignore [hunk 1]
```diff
@@ -6,6 +6,7 @@
 /yarn.lock
 /.pnp.*
 
+.swcrc
```
Comment (jlsnow301) [inline tgui/.prettierignore:9]: this is just fixing a prettier warn on swcrc

File: dependencies.sh [hunk 2]
```diff
@@ -12,7 +12,8 @@ export RUST_G_VERSION=3.1.0
 
 #node version
 export NODE_VERSION=14
-export NODE_VERSION_PRECISE=14.16.1
+export NODE_VERSION_PRECISE=20.12.0
+export NODE_VERSION_COMPAT=14.16.1
```
Comment (san7890) [inline dependencies.sh:16]: I think we should rename NODE_VERSION_PRECISE to be NODE_VERSION_LATEST or something like that given the new context we're using it in. PRECISE doesn't really make sense when the base `NODE_VERSION` is still `14`

Comment (jlsnow301) [reply dependencies.sh:16]: Can do

Comment (jlsnow301) [reply dependencies.sh]: not trying to be obtuse, "latest" has a different version in node, this is the LTS version and i don't want to confuse anyone familiar

Comment (san7890) [reply dependencies.sh]: ah okay

File: tools/bootstrap/node [hunk 3]
```diff
@@ -18,7 +18,7 @@ OldPWD="$PWD"
 cd "$Bootstrap/../.."
 . ./dependencies.sh  # sets NODE_VERSION_PRECISE
```
Comment (san7890) [inline tools/bootstrap/node:19]: probably should fix this comment now

**Pull request merged**